client/src/app/management/components/organisation-tag-dialog/organisation-tag-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+import { ComponentServiceCollector } from 'app/core/ui-services/component-service-collector';
+import { BaseComponent } from 'app/site/base/components/base.component';
+import { ViewOrganisationTag } from 'app/site/event-management/models/view-organisation-tag';
+
+interface OrganisationTagDialogData {
+    organisationTag?: ViewOrganisationTag;
+}
+
+@Component({
+    selector: 'os-organisation-tag-dialog',
+    templateUrl: './organisation-tag-dialog.component.html',
+    styleUrls: ['./organisation-tag-dialog.component.scss']
+})
+export class OrganisationTagDialogComponent extends BaseComponent implements OnInit {
+    public get isCreateView(): boolean {
+        return this._isCreateView;
+    }
+
+    public get currentColor(): string {
+        return this._lastValidColor;
+    }
+
+    public organisationTagForm: FormGroup;
+
+    private _isCreateView = false;
+    private _lastValidColor = '000';
+
+    public constructor(
+        serviceCollector: ComponentServiceCollector,
+        @Inject(MAT_DIALOG_DATA) public data: OrganisationTagDialogData,
+        private dialogRef: MatDialogRef<OrganisationTagDialogComponent>,
+        private fb: FormBuilder
+    ) {
+        super(serviceCollector);
+    }
+
+    public ngOnInit(): void {
+        this.createForm();
+        if (!this.data.organisationTag) {
+            this._isCreateView = true;
+        } else {
+            this.updateForm();
+        }
+    }
+
+    public onSaveClicked(): void {
+        const { name, color }: { name: string; color: string } = this.organisationTagForm.value;
+        this.dialogRef.close({ name, color: `#${color}` });
+    }
+
+    public generateColor(): void {
+        const singleValue = () => {
+            const nextValue = Math.round(Math.random() * 255).toString(16);
+            return nextValue.length === 2 ? nextValue : `0${nextValue}`;
+        };
+        const nextColor = [1, 1, 1].reduce(prevValue => prevValue + singleValue(), '');
+        this.organisationTagForm.patchValue({ color: nextColor });
+    }
+
+    private createForm(): void {
+        this.organisationTagForm = this.fb.group({
+            name: ['', Validators.required],
+            color: ['']
+        });
+        this.subscriptions.push(
+            this.organisationTagForm.get('color').valueChanges.subscribe((currentColor: string) => {
+                if (currentColor.length === 3 || currentColor.length === 6) {
```

Comment:
I know github allows 3 digit colors, but we do not.

Reply:
It is allowed, because HTML-colors can be define with three hex-digits, but okay - I will change this.